packages/shared/src/hooks/useRequestProtocol.ts

```diff
@@ -0,0 +1,21 @@
+import { useMemo } from 'react';
+import request from 'graphql-request';
+import { useQueryClient } from 'react-query';
+import { CompanionProtocol, COMPANION_PROTOCOL_KEY } from '../graphql/common';
+
+export const useRequestProtocol = (): {
+  requestMethod?: typeof request;
+  fetchMethod?: typeof fetch;
+} => {
+  const client = useQueryClient();
+  const { companionRequest, companionFetch } =
+    client.getQueryData<CompanionProtocol>(COMPANION_PROTOCOL_KEY) || {};
```

Comment:
For this hook.

I think it's better if the hook can simply can accept a request and fetch method, but we set it from the companion itself?

Now we kind of determine that here, but the companion should be the source of truth for which requestmethod it uses.

Not sure if this makes sense?

Reply:
Totally agree. Let me refactor this one.

Reply:
Amazing refactor, loving this 💖